src/NUnitFramework/framework/Exceptions/MultipleAssertException.cs

```diff
@@ -45,11 +45,15 @@ public class MultipleAssertException : ResultStateException
         /// used normally, when exiting the multiple assert block with failures.
         /// Not used internally but provided to facilitate debugging.
         /// </summary>
-        /// <param name="result">The current result</param>
-        public MultipleAssertException(ITestResult result)
-            : base(result.Message)
+        /// <param name="testResult">
+        /// The current result, up to this point. The result is not used 
+        /// internally by NUnit but is provided to facilitate debugging.
+        /// </param>
+        public MultipleAssertException(ITestResult testResult)
+            : base(testResult?.Message)
         {
-            TestResult = result;
+            Guard.ArgumentNotNull(testResult, "testResult");
```

Comment:
I'm still not convinced this is a good approach, because argument validation is being performed _after_ calling the base constructor. In these cases, inline static validation like what I posted earlier seems like the better way to go.

Reply:
Didn't see this one before merging. I considered creating some new utilities along the line of what you suggested to ensure the arg is not null but decided it was out of scope. The call to the base is made to be OK, but the derived constructor will then throw with a proper message. That seems good enough for now.